### PR TITLE
Add hidden cells and fix spelling

### PR DESCRIPTION
**Pull request by ma595**
The hidden cells are added in the first notebook to improve flow. The notebook is otherwise a little text-heavy. 

- Fixes typos to 01 and 03
- Adds more content / explanation to 01 where necessary.
- Gives user option to train on a reduced dataset in 03. 

TODO before merge:
- [x] Make optional performance improvement to 03 (if running locally)
- [ ] Make changes to solutions.

### CODE REVIEW

File: exercises/01_penguin_classification.ipynb [hunk 1]
```diff
@@ -434,51 +487,154 @@
...
+    "However, this might not be ideal for multi-class classification, where only one class should be correct.\n",
+    "In such cases, it’s usually better to use:\n",
+    "\n",
+    "CrossEntropyLoss\n",
+    "\n",
+    "Does not require one-hot encoding of targets.\n",
+    "\n",
+    "Combines softmax and negative log likelihood into a single, numerically stable operation.\n",
+    "\n",
+    "</details> "
```
Comment (jatkinson1000) [inline exercises/01_penguin_classification.ipynb:637]: I don't quite understand this. Sounds like it's maybe incomplete...?

File: exercises/01_penguin_classification.ipynb [hunk 2]
```diff
@@ -57,7 +57,10 @@
     "\n",
     "The penguin data reading and processing can be encapsulated in a PyTorch dataset class.\n",
     "\n",
-    "- Why is a class representation helpful?\n",
+    "\n",
+    "<details>\n",
+    "<summary>Why is a class representation helpful?</summary>\n",
+    "\n",
```
Comment (jatkinson1000) [inline exercises/01_penguin_classification.ipynb:63]: Some of the `...` were spaces to be completed by demonstrator/participants as they worked through the notebook. Are you still planning on doing this as part of the workshop, as if not these perhaps need completing (see worked solution for answers)?

File: exercises/01_penguin_classification.ipynb [hunk 3]
```diff
@@ -197,11 +202,14 @@
...
-    "            # print the features and targets here\n",
-    "        ```"
+    "  <details>\n",
+    "    <summary>Hint:</summary>\n",
+    "\n",
+    "    ```python\n",
+    "      for features, targets in dataset:\n",
+    "          # print the features and targets here\n",
+    "    ```\n",
+    "  </details>"
```
Comment (jatkinson1000) [inline exercises/01_penguin_classification.ipynb:212]: This 'hint' is repeated in the code cell below, so should probably just remove it from here.
```suggestion
```

File: exercises/01_penguin_classification.ipynb [hunk 4]
```diff
@@ -434,51 +487,154 @@
...
+    "from torchvision.models import resnet50\n",
+    "\n",
+    "# Create a ResNet50 model and print it.\n",
+    "model = resnet50(pretrained=True)\n",
+    "# Set the model to evaluation mode.\n",
+    "model.eval()\n",
+    "# Print the model architecture.\n",
+    "print(model)"
+   ]
+  },
```
Comment (jatkinson1000) [inline exercises/01_penguin_classification.ipynb:506]: In the original workshop we taught participants how to create a net from scratch and introduced loading pre-trained nets in a later exercise.

I see creating a net is still covered in task 6, so am not sure what this part is for?

File: exercises/01_penguin_classification.ipynb (hunk 2 above)
Comment (ma595) [reply exercises/01_penguin_classification.ipynb:63]: Yes, I agree. I left this question as a discussion point, but could add an ellipsis.

File: exercises/01_penguin_classification.ipynb [hunk 5]
```diff
@@ -434,51 +487,154 @@
...
-    "  - This activation function is good for classification when the result is one of ``A or B or C``.\n",
-    "    - It's bad if you even want to assign two classification to a single image—say a photo of a dog _and_ a cat.\n",
-    "  - It turns the raw outputs, or logits, into \"psuedo probabilities\", and we take our prediction to be the most probable class.\n",
+    "####  Choosing an Activation Function\n",
     "\n",
-    "- Have a go at writing these loops. Read the comments below for help.\n",
+    "Before we write our training and evaluation loops, we need to choose an **activation function** to apply to the model's outputs. This step ensures the outputs can be meaningfully compared to our targets — e.g. `[0, 0, 1]`.\n",
     "\n",
-    "TIPS:\n",
+    "> Note: We are **not** including the activation function inside the network architecture itself.\n",
```
[the remark this answers is not in the document]
Comment (ma595) [reply exercises/01_penguin_classification.ipynb:517]: Flexibility, performance. Should I expand on this?

[the remark this answers is not in the document]
Comment (ma595) [reply exercises/01_penguin_classification.ipynb:506]: Addressed review comment

[the remark this answers is not in the document]
Comment (jatkinson1000) [reply exercises/01_penguin_classification.ipynb:506]: Not a criticism, but more that you make this statement, but given this is a teaching module it would be good to explain to participants why.

File: exercises/01_penguin_classification.ipynb (hunk 1 above)
Comment (jatkinson1000) [reply exercises/01_penguin_classification.ipynb]: Yep, much better, thanks.

**Review by jatkinson1000 — COMMENTED** [review]
Thanks @ma595 

I have replied to a couple of comments, but will leave eventual decisions to yourself as the person delivering the course 😛 

Happy for the worked examples and conda branches to be updated later, but perhaps open an issue to track this and ensure it gets done as it's one of the key aspects of the workshop (and useful when delivering too, I found).

File: exercises/01_penguin_classification.ipynb (hunk 4 above)
Comment (ma595) [reply exercises/01_penguin_classification.ipynb]: Hmm, yes, I see the argument about overwriting a model with something else. This definitely needs to change. I will put the example in markdown and just have it there in true blue peter style (Here's one that was trained earlier). I'll change the variable `model` to something else so it doesn't corrupt the namespace.

Comment (ma595) [reply exercises/01_penguin_classification.ipynb]: My justification for including it was just to show this is how you can get a complicated model online and interrogate it's structure., to draw parallels with the simple one we've defined.
